### DISCUSSION TITLE
Integration with TestProject

### DISCUSSION THREAD

**Discussion by Alek5andr** [webdriverio/webdriverio#7559] [Ideas]
Hi!

Recently I've figured out such product like [TestProject](https://testproject.io/). The main necessity, that led me to it, was to find an ability to run iOS applications from Windows OS. It turned out that the product allows to run auto tests on connected via USB iOS devices from Windows. Unfortunately, Appium does not support it. As my research unfolded, it turned out that TestProject SDK supports Java, C# and Python languages. There is no **official** support of Node.js as I understood. However, I have found such [dependency](https://www.npmjs.com/package/@tpio/javascript-opensdk). Used technology to write test cases with this dependency seems to be outdated. _Or maybe it's just me, who sees it this way, because I practise WDIO 4 years and like it very much._
So my **idea proposition is to implement a service or somehow to integrate with TestProject**. I could not yet invent a solution on my own, how to combine WDIO technology with _@tpio/javascript-opensdk_ dependency.
If there would be people, who are interested in the idea, that would be an awesome support :)
P.S. Should I register a feature request?

**Comment by christian-bromann** [comment]
Hey @Alek5andr, thanks for starting the discussion.

It seems that running tests on TestProject only requires to set some [custom capabilities](https://github.com/testproject-io/javascript-opensdk/blob/7c4692216de1214a5a8497b25546f4b9adae8998/src/sdk/drivers/baseBuilder.ts#L272-L283) and specifying the testproject remote endpoint. Theoretically it should work already with WebdriverIO. I totally agree though that it could be better integrated, e.g. with a service as we have for Sauce Labs, Browserstack and other remote vendors. I am happy to add such a service to our service list and help with questions.

**Reply by Alek5andr** [reply]
Thank you @christian-bromann for the reply and pointing to TP driver's capabilities - I've not dived in so deep. Hence there is a lack of technical knowledge, how to use those without service integration yet. To have it as a service would be great.
Although TP requires its local server to be running to communicate with connected devices and TP's dashboard, I'll try to investigate overwriting of WDIO driver's capabilities further.
I'll let here know, if there is some success.
